pkg/dogstatsd/mapper/mapper.go

```diff
@@ -0,0 +1,124 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2019 Datadog, Inc.
+
+// Mapping feature is inspired by https://github.com/prometheus/statsd_exporter
+
+package mapper
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+)
+
+var (
+	allowedWildcardMatchPattern = regexp.MustCompile(`^[a-zA-Z0-9\-_*.]+$`)
+)
+
+const (
+	matchTypeWildcard = "wildcard"
+	matchTypeRegex    = "regex"
+)
+
+// MetricMapper contains mappings and cache instance
+type MetricMapper struct {
+	Mappings []MetricMapping
+	cache    *mapperCache
+}
+
+// MetricMapping represent one mapping rule
+type MetricMapping struct {
+	Match     string            `mapstructure:"match"`
+	MatchType string            `mapstructure:"match_type"`
+	Name      string            `mapstructure:"name"`
+	Tags      map[string]string `mapstructure:"tags"`
+	regex     *regexp.Regexp
+}
+
+// NewMetricMapper creates, validates, prepares a new MetricMapper
+func NewMetricMapper(configMappings []MetricMapping, cacheSize int) (MetricMapper, error) {
```

Comment:
Nit: returning a pointer is more idiomatic and you convert it a few lines after this is called.

Reply:
Sure, done.

Reply:
Sure, done.